### PR TITLE
Add ability to customize default implementation of bonded<T> in C#

### PR DESCRIPTION
**Pull request by alexsatch**

### CODE REVIEW

File: cs/test/core/Core.csproj [hunk 1]
```diff
@@ -1,116 +1,117 @@
...
-    <Reference Include="Bond.JSON">
-      <HintPath>..\..\src\json\$(DependentOutputPath)\Bond.JSON.dll</HintPath>
-    </Reference>
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
-  </ItemGroup>
-  <Import Project="$(MSBuildThisFileDirectory)\..\..\build\Common.Internal.targets" />
-</Project>
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (lalo) [inline cs/test/core/Core.csproj]: line ending issue?

**Comment by chwarr** [comment]
Can you please rebase the commits to get rid of the work-in-progress and fix up commits? It seems like we should be able to "tell the story" of this change in one or two commits.

**Comment by chwarr** [comment]
Travis CI build errors are ones we're seeing in #145 as well. We're still trying to figure out what's going on to trigger this.

Test result excerpts:
```
Test Error : UnitTest.Aliases.TypeAliasTests.AliasBlob
   System.ArgumentNullException : Value cannot be null.
Parameter name: method
  at System.Linq.Expressions.Expression.Call (System.Linq.Expressions.Expression instance, System.Reflection.MethodInfo method, IEnumerable`1 arguments) <0x405b8080 + 0x001c3> in <filename unknown>:0 
  at System.Linq.Expressions.Expression.Call (System.Linq.Expressions.Expression instance, System.Reflection.MethodInfo method) <0x405b8030 + 0x0002f> in <filename unknown>:0 
  at Bond.Expressions.UntaggedReader`1[R].ReadFieldOmitted () <0x40762bf0 + 0x0006b> in <filename unknown>:0 
  at Bond.Expressions.UntaggedParser`1[R].Field (ITransform transform, Bond.FieldDef fieldDef, IField field) <0x40762860 + 0x001bd> in <filename unknown>:0 
  at Bond.Expressions.UntaggedParser`1+<Apply>c__AnonStorey0[R].<>m__0 (<>__AnonType1`2 <>__TranspIdent1, IField knownField) <0x40762780 + 0x00097> in <filename unknown>:0 
  at System.Linq.Enumerable+<SelectManyIterator>c__Iterator5`3[TSource,TCollection,TResult].MoveNext () <0x40601d10 + 0x002da> in <filename unknown>:0 
  at System.Collections.Generic.List`1[T].InsertRange (Int32 index, IEnumerable`1 collection) <0x400dd4c0 + 0x002a8> in <filename unknown>:0 
  at System.Collections.Generic.List`1[T].AddRange (IEnumerable`1 collection) <0x400dd480 + 0x00025> in <filename unknown>:0 
  at Bond.Expressions.UntaggedParser`1[R].Apply (ITransform transform) <0x40761a30 + 0x00a8b> in <filename unknown>:0 
  at Bond.Expressions.DeserializerTransform`1[R].Struct (IParser parser, System.Linq.Expressions.Expression var, System.Type schemaType, Boolean initialize) <0x406b4000 + 0x00664> in <filename unknown>:0 
  at Bond.Expressions.DeserializerTransform`1[R].Deserialize (IParser parser, System.Linq.Expressions.Expression var, System.Type objectType, System.Type schemaType, Boolean initialize) <0x406a3250 + 0x004f5> in 
```

**Comment by alexsatch** [comment]
I will redo the PR with squashed commits. Trying to fix the line ending issue in Core.csproj

**Comment by chwarr** [comment]
See #153 for the continuation of this PR.

**Pull request closed without merging**